exonum/tests/proof_map_index_prop.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright 2018 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// cspell:ignore oneof
+
+//! Property testing for proof map index as a rust collection.
+
+extern crate exonum;
+#[macro_use]
+extern crate proptest;
+
+use exonum::storage::{Database, Fork, MemoryDB, ProofMapIndex};
+use proptest::{collection::vec, num, prelude::*, strategy};
+
+use std::collections::HashMap;
+
+#[derive(Debug, Clone)]
+enum Action {
+    //Applied to key where key[0] is modulo 8
```

Comment:
Please use spaces after `//`, full stops at the end and backticks where appropriate.